tests/snapshots/encode_pattern_match__common.hvm.snap

```diff
@@ -15,6 +15,10 @@ TaggedScott:
 
 (Yellow) = #Light λ* #Light λb #Light λ* b
 
+(Red) = #Light λa #Light λ* #Light λ* a
```

Comment:
Did we change something to cause these to have a different order? Might have something causing the output to not be deterministic across supposedly equivalent versions of the compiler.

Reply:
It is related with the `data Result` in the builtins.hvm file, without `Result` it outputs in correct order

Reply:
I don't understand why it changes the order of other unrelated things. It should be in insertion order here, right? I would understand if it were after the merge pass, since it swaps things around.